### PR TITLE
Don't emit ANSI colour escape sequences when output is not a tty

### PR DESCRIPTION
**Pull request by ldrumm**
When piping the output of utest.h into a file or pipe the default
behaviour should be to omit colour formatting as these output devices
are non-interactive and colour output confounds parsing.

In a future patch I'd like to add support to make this optional a'la

git diff --color=<auto|always|never> or gcc's
`-fdiagnostics-color=<auto|always|never>`

However, this should be sufficient for now.

### CODE REVIEW

File: utest.h [hunk 1]
```diff
@@ -154,6 +154,27 @@
 #define UTEST_EXTERN extern
 #endif
 
+#ifdef _WIN32
```
Comment (sheredom) [inline utest.h]: I only use _MSC_VER to detect windows - any reason why you used _WIN32 here?

Comment (ldrumm) [reply utest.h]: `_MSC_VER` only refers to MSVC which is the compiler causing the warnings, and supports these pragmas. gcc(mingw) also sets `_WIN32` (and needs io.h included) but shouldn't have the `#pragma` block.

Comment (ldrumm) [reply utest.h]: If use of non-msvc compilers on windows is an explicit non-goal of the project I guess `#ifdef _MSC_VER` is enough - no need for the nested `#if`. However, if you do intend to support gcc on windows I think it needs to be left the way it is.

Comment (sheredom) [reply utest.h]: So given that I have no testing (and I don't think any way TO test) on Windows that is non MSVC, I'd rather remove it and just use _MSC_VER - I don't want people thinking that there is support when there definitely isn't.

Comment (ldrumm) [reply utest.h]: Agreed. Will fix.

**Pull request merged**